src/odemis/acq/stream/_sync.py

```diff
@@ -1371,22 +1378,28 @@ def _runAcquisitionEbeam(self, future):
 
     def _integrateImages(self, n):
         """
-        Integrate/sum the last n acquired images in self._acq_data and replace by the
-        integrated image. If values overflow, clip the image. If number of images = 1,
-        no image integration necessary.
-        :param n: (int) Number of images that need to be integrated (summed).
+        Integrate/sum the images in self._acq_data. If number of images = 1, no image integration necessary.
+        If number of images > 10, integrate one image after another and replace by the integrated image.
+        It's about integrating the new acquired image in self._acq_data with the last integrated image if exists.
+        If number of images <= 10, integrate/sum the last n acquired images in self._acq_data and replace by the
+        integrated image. If values overflow, clip the image.
+        Args:
+            n: (int) Number of images that need to be integrated (summed).
```

Comment:
I think it should be simplified and moved to `odemis.util.img` as `sum_img(imgs, init=None)`. imgs is a list of images. init is a single image. It returns a single image. If there is no init, it computes the right dtype to ensure it cannot overflow. If init is present, it uses it to store the result, and so its dtype should fit, and it will be also the return value.

In the stream, here you "only" have to call this function correctly. When receiving the first image, compute the right dtype with `get_best_dtype_for_acc(orig_dtype, n)`, and convert the image. Then, for every other image, call `sum_img()`.

Reply:
I did it in the _sync.py because I wanted to replace the images saved in self._acq_data so as the self._acq_data is not getting too big.